activesupport/test/digest_test.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require "abstract_unit"
+require "openssl"
+
+class DigestTest < ActiveSupport::TestCase
+  class InvalidDigest; end
+
+  def test_with_default_hash_digest_class
+    assert_equal ::Digest::MD5.hexdigest("hello friend"), ActiveSupport::Digest.hexdigest("hello friend")
+  end
+
+  def test_with_custom_hash_digest_class
+    assert_equal ::Digest::SHA1.hexdigest("hello friend"),
+                 ActiveSupport::Digest.new(digest_class: ::Digest::SHA1).hexdigest("hello friend")
```

Comment:
We should change the global configuration and call `ActiveSupport::Digest.hexdigest` here, since that's the code path that the rest of the framework goes through:

```ruby
def test_with_custom_hash_digest_class
  old_hash_digest_class = ActiveSupport::Digest.hash_digest_class
  ActiveSupport::Digest.hash_digest_class = ::Digest::SHA1
  assert_equal ::Digest::SHA1.hexdigest("hello friend"), ActiveSupport::Digest.hexdigest("hello friend")
ensure
  ActiveSupport::Digest.hash_digest_class = old_hash_digest_class
end
```